internal/convenience/repository/report_repository.go

```diff
@@ -209,30 +219,17 @@ func (c *ReportRepository) FindAll(
 	}
 	defer stmt.Close()
 
-	var reports []cModel.Report
-	rows, err := stmt.QueryxContext(ctx, args...)
+	var rows []reportRow
+	err = stmt.SelectContext(ctx, &rows, args...)
+
 	if err != nil {
+		slog.Error("Find all error", "error", err)
 		return nil, err
 	}
-	defer rows.Close()
+	reports := make([]cModel.Report, len(rows))
```

Comment:
Parece que o defer rows.Close() foi retirado

Reply:
Diego, it's not necessary to call row.Close() here because we're using stmt.SelectContext() to execute the query and directly populate the rows structure. Since the rows object doesn't maintain an open connection to the database, defer stmt.Close() is enough to handle it.